src/rest-client.js

```diff
@@ -136,6 +137,13 @@ function resource(client, parent, name, id, ctx) {
             if (shortcut) {
                 self._shortcuts[resName] = r;
                 self[resName] = r;
+                client._opts.shortcutRules.forEach(rule => {
+                    let customShortcut = rule.call(undefined, resName);
```

Comment:
Why not just `rule(resName)`?

Reply:
It was intended to lock access to the object. But this should also "break" the this-binding of arrow functions. Good spot, I will replace this line.